src/components/lab/SpectrumBar.js

```diff
@@ -18,16 +19,47 @@ import {
   ULTRAVIOLET_BAR_LABEL_COLOR,
   WAVELENGTH_LABELS_X_AXIS_ADJUSTMENT_FACTOR,
   WAVELENGTH_LABELS_Y_AXIS_ADJUSTMENT_FACTOR,
+  MEASURING_ARROW_UNITS_TO_NANOMETER_CONVERSION_FACTOR,
+  MAX_DISPLAYED_WAVELENGTH,
+  MIN_DISPLAYED_WAVELENGTH,
+  APPROXIMATE_WAVELENGTH_LABEL_WIDTH,
 } from '../../config/constants';
+import { calculateWavelength } from '../../utils/physics';
 
-const SpectrumBar = ({ stageWidth, stageHeight }) => {
+const SpectrumBar = ({
+  stageWidth,
+  stageHeight,
+  frequency,
+  shouldOscillate,
+}) => {
+  // ref attached to wavelength label (used to detect its width and position it)
+  // initialized to APPROXIMATE_WAVELENGTH_LABEL_WIDTH to minimize adjustment/jump after element has been rendered
+  const wavelengthLabel = useRef(null);
+  const [wavelengthLabelWidth, setWavelengthLabelWidth] = useState(
+    APPROXIMATE_WAVELENGTH_LABEL_WIDTH,
+  );
   const { t } = useTranslation();
 
+  // on component mount, measure width of wavelength label, update state
+  useEffect(() => {
+    setWavelengthLabelWidth(wavelengthLabel.current.textWidth);
+  }, []);
+
   // centers spectrum bar horizontally
   const spectrumBarInitialXPosition =
     stageWidth / 2 - TOTAL_SPECTRUM_BAR_WIDTH / 2;
   const spectrumBarInitialYPosition = 0.85 * stageHeight;
 
+  // used to determine x position of SpectrumBarMarker; if wavelength > 1000 or < 100, return 0 (no spectrum bar shown)
+  const wavelength = calculateWavelength(frequency);
+  const spectrumBarMarkerXPosition =
+    wavelength > MAX_DISPLAYED_WAVELENGTH ||
+    wavelength < MIN_DISPLAYED_WAVELENGTH
+      ? 0
+      : spectrumBarInitialXPosition +
+        (1000 - wavelength) /
```

Comment:
Would it be possible to change this 1000 by the max displayed wavelength?

Reply:
👍